provider/godaddy/client.go

```diff
@@ -292,8 +311,8 @@ func (c *Client) CallAPI(method, path string, reqBody, resType interface{}, need
 //
 // If everything went fine, unmarshall response into resType and return nil
 // otherwise, return the error
-func (c *Client) CallAPIWithContext(ctx context.Context, method, path string, reqBody, resType interface{}, needAuth bool) error {
-	req, err := c.NewRequest(method, path, reqBody, needAuth)
+func (c *Client) CallAPIWithContext(ctx context.Context, method, path string, reqBody, resType interface{}) error {
+	req, err := c.NewRequest(method, path, reqBody)
```

Comment:
@alexstojda Would you please detail why you are removing the `needAuth` boolean which is set to true on all call ?

Reply:
My local linter complained about this as the variable is unused in the function.